### ISSUE TITLE
https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:relocation

### ISSUE THREAD

**Issue by xenoterracide** [gradle/gradle#31044]
### Issue type

Wrong or misleading information

### Problem description

relocation docs use create instead of register

### Page with the problem

https://docs.gradle.org/current/userguide/publishing_maven.html#publishing_maven:relocation

**Comment by xenoterracide** [comment]
I don't know why they use create.  Okay I'm probably being a smart butt.  I assume that your question is why it should change. From what I can tell by the Java doc the register function is the lazy variant of create. I am of the opinion that the lazy variants should be recommended before the non lazy ones. Be lazy unless you have a reason not to be.

**Comment by abstratt** [comment]
This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

***

There seems to be an inconsistency w.r.t. `create`/`register` between task containers and other kinds of containers. For instance, registering a publication will cause publication tasks not to be registered.

**Comment by xenoterracide** [comment]
I know they aren't effectively lazy... I've assumed that was just a detail. Are you saying using it actually breaks something?

**Comment by big-guy** [comment]
Yes, depending on the container, using register vs create can cause configuration to be silently ignored unless the container or element is realized.  For a lot of dependency management containers, we end up realizing everything so it doesn't necessarily matter.

**Comment by xenoterracide** [comment]
Is that actually the case here? I mean the artifact seems to be created in my case so I'm kind of making the assumption that it's created correctly. 

It's definitely sounds like a reason why there shouldn't be two apis; regardless of whether an API is actually used under the hood eagerly or lazily. Of course lazy should be preferred generally speaking.

**Comment by big-guy** [comment]
Yes, we have several places where we end up looking at the publication container in an eager way that makes create/register behave the same in the end.

The issues with two different APIs on the container is a cross-cutting issue and we're not likely to make any changes before 9.0 and that needs to be settled across all the different containers (not just the publications one). 

Docs wise, I think it's worse to use register right now because it makes it seem like you should use register everywhere.
